artifacthub/library/general/storageclass/1.1.0/template.yaml

```diff
@@ -0,0 +1,155 @@
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sstorageclass
+
+        is_pvc(obj) {
+          obj.apiVersion == "v1"
+          obj.kind == "PersistentVolumeClaim"
+        }
+
+        is_statefulset(obj) {
+          obj.apiVersion == "apps/v1"
+          obj.kind == "StatefulSet"
+        }
+
+        violation[{"msg": msg}] {
+          not data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"]
+          msg := sprintf("StorageClasses not synced. Gatekeeper may be misconfigured. Please have a cluster-admin consult the documentation.", [])
+        }
+
+        storageclass_allowed(name) {
+          data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][name]
+          # support both direct use of * and as the default value
+          object.get(input.parameters, "allowedStorageClasses", ["*"])[_] == "*"
+        }
+
+        storageclass_allowed(name) {
+          data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][name]
+          input.parameters.allowedStorageClasses[_] == name
+        }
+
+        violation[{"msg": pvc_storageclass_badname_msg}] {
+          is_pvc(input.review.object)
+          not storageclass_allowed(input.review.object.spec.storageClassName)
+        }
+        pvc_storageclass_badname_msg := sprintf("pvc did not specify a valid storage class name <%v>. Must be one of [%v]", args) {
+          input.parameters.includeStorageClassesInMessage
+          object.get(input.parameters, "allowedStorageClasses", null) == null
+          args := [
+            input.review.object.spec.storageClassName,
+            concat(", ", [n | data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][n]])
+          ]
+        } else := sprintf("pvc did not specify an allowed and valid storage class name <%v>. Must be one of [%v]", args) {
+          input.parameters.includeStorageClassesInMessage
+          object.get(input.parameters, "allowedStorageClasses", null) != null
+          sc := {n | data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][n]} & {x | x = object.get(input.parameters, "allowedStorageClasses", [])[_]}
+          args := [
+            input.review.object.spec.storageClassName,
+            concat(", ", sc)
+          ]
+        } else := sprintf(
+          "pvc did not specify a valid storage class name <%v>.",
+          [input.review.object.spec.storageClassName]
+        )
+
+        violation[{"msg": pvc_storageclass_noname_msg}] {
+          is_pvc(input.review.object)
+          not input.review.object.spec.storageClassName
+        }
+        pvc_storageclass_noname_msg := sprintf("pvc did not specify a storage class name. Must be one of [%v]", args) {
+          input.parameters.includeStorageClassesInMessage
+          args := [
+            concat(", ", [n | data.inventory.cluster["storage.k8s.io/v1"]["StorageClass"][n]])
+          ]
+        } else := sprintf(
```

Comment:
nit: does the syntax allow just defining a string constant here?

Reply:
Hey @maxsmythe 

This was existing code, so I was also a bit surprised by the `variable := string { } else := "string"` syntax, but it appears to work as expected and match the Rego doc: https://www.openpolicyagent.org/docs/latest/policy-language/#else-keyword